cloud/defaults.go

```diff
@@ -40,11 +41,28 @@ const (
 	LatestVersion = "latest"
 )
 
+const (
+	// AKSSubnetName is the name of the default aks managed subnet for worker nodes.
+	AKSSubnetName = "aks-subnet"
+	// UniqueNameSuffixSize is the length of the hash AKS uses as a cluster id.
+	UniqueNameSuffixSize = 8
+)
+
 // GenerateInternalLBName generates a internal load balancer name, based on the cluster name.
 func GenerateInternalLBName(clusterName string) string {
 	return fmt.Sprintf("%s-%s", clusterName, "internal-lb")
 }
 
+// GenerateManagedResourceGroupName returns the name of the resource group containing IaaS resources for an AKS cluster.
+func GenerateManagedResourceGroupName(resourceGroup, clusterName, location string) string {
```

Comment:
instead of assuming the default format here, would it make sense to get the nodeResourceGroup value from ManagedClusterProperties ?

Reply:
good point, I forgot it doesn't technically have to be the MC group